ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/healthcheck/RangerHealthCheck.java

```diff
@@ -19,18 +19,22 @@
 import lombok.extern.slf4j.Slf4j;
 import org.apache.curator.framework.CuratorFramework;
 
+import java.util.List;
+
 @Slf4j
 public class RangerHealthCheck extends HealthCheck {
 
-  private final CuratorFramework curatorFramework;
+  private final List<CuratorFramework> curatorFrameworks;
```

Comment:
Why this is needed? We can keep only one CuratorFramework?

Reply:
Even I am confused, this is to ensure that a container is connected to it's relevant ZK .. why is this a list now?

Reply:
All concrete implementations of `RangerUpstreamConfiguration` are having list of upstreams of same type - HTTP / ZK /DROVE. `RangerHubServerBundle` initializes a list of `curatorFrameworks` and we need a single `RangerHealthCheck` for all those `curatorFrameworks`.

When registering individual `RangerHealthCheck` for each `curatorFramework` , it throws an error because healthcheck name is same - simple class name for `RangerHealthCheck`.

We can either change  `RangerUpstreamConfiguration` to only take config of single upstream or register individual `RangerHealthCheck` for each `curatorFramework` with a different name. Please consider if this refactoring can be done in a new pull request